iquip/apps/monitor.py

```diff
@@ -619,25 +619,265 @@ def _setSwitchButtonText(self, on: bool):
         self.switchClicked.emit(on)
 
 
-class DeviceMonitorApp(qiwis.BaseApp):
-    """App for monitoring and controlling ARTIQ hardwares e.g., TTL, DDS, and DAC.
+class DDSControllerFrame(QWidget):
+    """Frame for monitoring and controlling DDS channels.
+    
+    Attributes:
+        ddsWidgets: Dictionary with DDS controller widgets.
+          Each key is a DDS channel name, and its value is the corresponding DDSControllerWidget.
+    """
+
+    def __init__(
+        self,
+        ddsInfo: Dict[str, Dict[str, Any]],
+        numColumns: int = 4,
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+        
+        Args:
+            ddsInfo: Dictionary with DDS channels info.
+              Each key is a DDS channel name, and its value is a dictionary with DDS info.
+              This dictionary is given as keyword arguments to DDSControllerWidget.__init__().
+            numColumns: Number of columns in DDS widgets container layout.
+        """
+        super().__init__(parent=parent)
+        if numColumns <= 0:
+            logger.error("The number of columns must be positive.")
+            return
+        self.ddsWidgets: Dict[str, DDSControllerWidget] = {}
+        # widgets
+        ddsWidgetLayout = QGridLayout()
+        for idx, (name, info) in enumerate(ddsInfo.items()):
+            ddsWidget = DDSControllerWidget(name, **info)
+            row, column = idx // numColumns, idx % numColumns
+            self.ddsWidgets[name] = ddsWidget
+            ddsWidgetLayout.addWidget(ddsWidget, row, column)
+        # layout
+        layout = QVBoxLayout(self)
+        layout.addLayout(ddsWidgetLayout)
```

Comment:
Why don't you just set the grid layout as the layout of `self`?

Reply:
I added the outermost layout for unity with `TTLControllerFrame` and `DACControllerFrame`.
Also, several widgets may be added later, such as description labels, so I want to remain it as it is.